ArtOfIllusion/src/artofillusion/Scene.java

```diff
@@ -1208,8 +1209,59 @@ public Scene(DataInputStream in, boolean fullScene) throws IOException, InvalidO
     initFromStream(in, fullScene);
   }
 
+  @SuppressWarnings("UseOfObsoleteCollectionType")
+  private static Vector<Named> loadSceneAsset(DataInputStream source, Scene scene, Supplier<Named> stub) throws IOException
+  {
+      int counter = source.readInt();
+      Vector<Named> result = new Vector<>(counter);
+
+      for (int i = 0; i < counter; i++)
+      {
+        String className;
+        byte[] bytes;
+
+        //Unrecoverable IOException may thrown from next block..
+        {
+          className = source.readUTF();
+          bytes = new byte[source.readInt()];
+          source.readFully(bytes);
+        }
+        //Now lookup class by Name. Process ClassNotFound Exception as recovarable error and add stub asset to result
+        Class<?> clazz;
+        try
+        {
+          clazz = ArtOfIllusion.getClass(className);
+        }
+        catch(ClassNotFoundException cnfe) //Exception thrown once class not found across loaded plugins...
+        {
+          result.add((Named)stub.get().setNameFluent("<unreadable>"));
+          scene.errors.add(Translate.text("errorFindingClass", className));
+          continue;
+        }
+        if(clazz == null) // Null class returns from ArtofIllusion.getClass when no loaded plugins at all
+        {
+          result.add((Named)stub.get().setNameFluent("<unreadable>"));
+          scene.errors.add(Translate.text("errorFindingClass", className));
+          continue;
+        }
+        try
+        {
+          Constructor<?> ctor = clazz.getConstructor(DataInputStream.class, Scene.class);
+          InputStream byteStream = new DataInputStream(new ByteArrayInputStream(bytes));
+          result.add((Named)ctor.newInstance(byteStream, scene));
+        }
+        catch(IllegalAccessException | IllegalArgumentException | InstantiationException | NoSuchMethodException | SecurityException | InvocationTargetException ex)
+        {
+          result.add((Named)stub.get().setNameFluent("<unreadable>"));
+          scene.errors.add(Translate.text("errorInstantiatingClass", className));
+        }
+      }
+      return result;
+  }
+
   /** Initialize the scene based on information read from an input stream. */
 
+  @SuppressWarnings("UseOfObsoleteCollectionType")
```

Comment:
Whatever else we decide about this change, please reserve `@SuppressWarnings` for instances where we *cannot* change to a newer collection class/structure/safe cast/etc.

And even then, It should probably go at the line that uses the 'iffy' code, with a comment that explains *why* we have to stick to the old way...